Muehle_Algo.ipynb

```diff
@@ -245,29 +341,62 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 12,
    "metadata": {},
    "outputs": [],
    "source": [
-    "def best_move_ab(state, player, depth):\n",
-    "    ns     = next_states(state, player)\n",
-    "    moves  = []\n",
-    "    values = []\n",
-    "    ### iterative deepening\n",
-    "    for s in ns:\n",
-    "        distance = 1\n",
-    "        while distance <= (depth - 1):\n",
-    "            value = -value_ab(s, opponent(player), depth = distance)\n",
-    "            distance += 1\n",
-    "        values = values + [value]\n",
-    "        moves = moves + [s]\n",
-    "    ###\n",
-    "    best_value = max(values)\n",
-    "    best_moves = []\n",
-    "    for i in range(0,len(values)):\n",
-    "        if values[i] == best_value:\n",
-    "            best_moves = best_moves + [moves[i]]\n",
-    "    best_state = random.choice(best_moves)\n",
+    "def best_move_ab_old(state, player, depth):\r\n",
+    "    ns     = next_states(state, player)\r\n",
+    "    moves  = []\r\n",
+    "    values = []\r\n",
+    "    ### iterative deepening\r\n",
+    "    for s in ns:\r\n",
+    "        distance = 1\r\n",
+    "        while distance <= (depth - 1):\r\n",
+    "            value = -value_ab(s, opponent(player), depth = distance)\r\n",
+    "            distance += 1\r\n",
+    "        values = values + [value]\r\n",
+    "        moves = moves + [s]\r\n",
+    "    ###\r\n",
+    "    best_value = max(values)\r\n",
+    "    best_moves = []\r\n",
+    "    for i in range(0,len(values)):\r\n",
+    "        if values[i] == best_value:\r\n",
+    "            best_moves = best_moves + [moves[i]]\r\n",
+    "    best_state = random.choice(best_moves)\r\n",
+    "    return best_value, best_state"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from operator import itemgetter\r\n",
+    "import random\r\n",
+    "\r\n",
+    "def best_move_ab(state, player, depth):\r\n",
+    "    ns     = next_states(state, player)\r\n",
+    "    #in tupel-liste einfügen\\n\",\r\n",
+    "    nextStates = []\r\n",
```

Comment:
nextStates -> next_states (bzw. dann anderen Namen überlegen, weil Überschneidung mit Funktionsnamen)
In der Doku erklären, was der unterschied zwischen ns und nextStates ist

Reply:
Hier hast Du Dir noch nichts überlegt